lib/braveAlerter.js

```diff
@@ -346,26 +379,76 @@ class BraveAlerter {
       if (alertSession === null) {
         const errorMessage = `Failed to acknowledge alert for session ${sessionId}: No corresponding session`
         helpers.logError(errorMessage)
-        response.status(400).send(errorMessage)
+        response.status(400).json(errorMessage)
         return
       }
 
       if (alertSession.alertState !== CHATBOT_STATE.STARTED && alertSession.alertState !== CHATBOT_STATE.WAITING_FOR_REPLY) {
-        // This case is unexpected, but could potentially happen if the app allowed a double-click on the ACK button
+        // This case is unexpected, but could potentially happen if the app allowed a double-click on the button
         // So we'll log it because it's unnecessary, but still return 200 because it probably isn't wrong
         helpers.log(
           `Failed to acknowledge alert for session ${sessionId}: Session has already been acknowledged (current state: ${alertSession.alertState})`,
         )
-        response.status(200).send()
+      } else {
+        const changedAlertSession = new AlertSession(alertSession.sessionId, CHATBOT_STATE.RESPONDING)
+        await this.alertSessionChangedCallback(changedAlertSession)
+      }
+
+      const activeAlerts = await this.getActiveAlertsByAlertApiKey(alertApiKey)
+      if (activeAlerts) {
+        response.status(200).json(JSON.stringify(activeAlerts))
+      } else {
+        response.status(200).json(JSON.stringify({}))
+      }
+    } catch (err) {
+      const errorMessage = `Failed to acknowledge alert for session ${sessionId}: ${err.toString()}`
+      helpers.logError(errorMessage)
+      response.status(500).json(errorMessage)
+    }
+  }
+
+  async handleRespondToAlertSession(request, response) {
```

Comment:
This is called off of a button on the push notification (or a button in the app when it's in the foreground?). And this updates the session in the db to have the Responding state, is that right?

Reply:
Exactly!